### DISCUSSION TITLE
Okta auth got `read ECONNRESET`

### DISCUSSION THREAD

**Discussion by tmwatchanan** [langfuse/langfuse#1601] [Support]
### Describe the bug

I just tried Okta auth0 in my self-hosted Langfuse (v2.15.3). But I encountered this error. Any idea how to solve it? I can still use email/pwd authentication without any issue. I've set the following:

```
[next-auth][error][SIGNIN_OAUTH_ERROR] 
https://next-auth.js.org/errors#signin_oauth_error read ECONNRESET {
  error: {
    message: 'read ECONNRESET',
    stack: 'Error: read ECONNRESET\n' +
      '    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20)',
    name: 'Error'
  },
  providerId: 'okta',
  message: 'read ECONNRESET'
}
```

What I noticed on the client side is that it seems to make a request to something like this (a bunch of duplicate callbackUrl to <my-domain.com>'s)

> /auth/sign-in?callbackUrl=https%3A%2F%2F<my-domain.com>%2Fauth%2Fsign-in%3FcallbackUrl%3Dhttps%253A%252F%252F<my-domain.com>%252Fauth%252Fsign-in%253FcallbackUrl%253Dhttps%25253A%25252F%25252F<my-domain.com>%25252Fauth%25252Fsign-in%25253FcallbackUrl%25253Dhttps%2525253A%2525252F%2525252F<my-domain.com>%2525252Fauth%2525252Fsign-in%2525253FcallbackUrl%2525253Dhttps%252525253A%252525252F%252525252F<my-domain.com>%252525252Fauth%252525252Fsign-in%252525253FcallbackUrl%252525253Dhttps%25252525253A%25252525252F%25252525252F<my-domain.com>%25252525252Fauth%25252525252Fsign-in%25252525253FcallbackUrl%25252525253Dhttps%2525252525253A%2525252525252F%2525252525252F<my-domain.com>%2525252525252Fauth%2525252525252Fsign-in%2525252525253FcallbackUrl%2525252525253Dhttps%252525252525253A%252525252525252F%252525252525252F<my-domain.com>%252525252525252Fauth%252525252525252Fsign-in%252525252525253FcallbackUrl%252525252525253Dhttps%25252525252525253A%25252525252525252F%25252525252525252F<my-domain.com>%25252525252525252Fauth%25252525252525252Fsign-in%2525252525252526error%252525252525253DOAuthSignin%25252525252526error%2525252525253DOAuthSignin%252525252526error%25252525253DOAuthSignin%2525252526error%252525253DOAuthSignin%25252526error%2525253DOAuthSignin%252526error%25253DOAuthSignin%2526error%253DOAuthSignin%26error%3DOAuthSignin&error=OAuthSignin

### To reproduce

Set these environment variables:
- AUTH_OKTA_CLIENT_ID
- AUTH_OKTA_CLIENT_SECRET
- AUTH_OKTA_ISSUER
- NEXTAUTH_SECRET
- NEXTAUTH_URL is my domain https://my-domain.com/

**Comment by marcklingen** [comment]
Langfuse uses Auth.js to integrate with Okta. One common issue is that `NEXTAUTH_URL` is not setup correctly leading to all sorts of SSO issues.

Docs: https://langfuse.com/docs/deployment/self-host#configuring-environment-variables

**Comment by tmwatchanan** [comment]
I've already set `NEXTAUTH_URL` to be my domain. But all I can get is the above error. Do you happen to know any other things to check with?

**Reply by marcklingen** [reply]
no obvious idea here, open for PRs on this in case you find a fix for this issue or if we can improve the docs regarding the Okta integration. We use very much plain vanilla NextAuth/Auth.js <> Okta

**Reply by tmwatchanan** [reply]
Thanks! I'll open a PR if I find a solution to this.

**Reply by marcklingen** [reply]
Thanks 🙏

**Comment by victor-at-square** [comment]
@tmwatchanan Were you able to diagnose the cause here? We're observing a similar problem.

**Reply by tmwatchanan** [reply]
You have to configure Okta's sign-in redirect URI to be like `https://<your-langfuse-domain>/api/auth/callback/okta`.

**Reply by victor-at-square** [reply]
Thanks for the quick response. That looks right to me.

I think ours might be a different issue -- potentially a proxy / firewall.

**Reply by tmwatchanan** [reply]
Try to check that. Hope you solve it

**Comment by Colstuwjx** [comment]
Another issue I've found is that: you must set AUTH_OKTA_CHECKS=state otherwise it would report "The authentication request has an invalid 'state' parameter." and can't sign in any more.

**Reply by SebastianUrbanoL** [reply]
Thanks for this comment, it worked for me. I had this issue for future references 

error_description: "The authentication request has an invalid 'state' parameter.",
providerId: 'okta',
message: 'invalid_request'
}
[next-auth][error][OAUTH_CALLBACK_ERROR]
https://next-auth.js.org/errors#oauth_callback_error invalid_request {
message: 'invalid_request',
stack: 'Error: invalid_request\n' +
' at oAuthCallback (/app/node_modules/.pnpm/next-auth@4.24.11_patch_hash=cczsf6i6qe2m2htirvgxjtoclu_next@14.2.30_@opentelemetry+api@1.9.0_schh4asta4bbq2ii25cs3g5pay/node_modules/next-auth/core/lib/oauth/callback.js:29:19)\n' +
' at Object.callback (/app/node_modules/.pnpm/next-auth@4.24.11_patch_hash=cczsf6i6qe2m2htirvgxjtoclu_next@14.2.30_@opentelemetry+api@1.9.0_schh4asta4bbq2ii25cs3g5pay/node_modules/next-auth/core/routes/callback.js:45:39)\n' +
' at AuthHandler (/app/node_modules/.pnpm/next-auth@4.24.11_patch_hash=cczsf6i6qe2m2htirvgxjtoclu_next@14.2.30_@opentelemetry+api@1.9.0_schh4asta4bbq2ii25cs3g5pay/node_modules/next-auth/core/index.js:175:41)\n' +
' at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
' at async NextAuthApiHandler (/app/node_modules/.pnpm/next-auth@4.24.11_patch_hash=cczsf6i6qe2m2htirvgxjtoclu_next@14.2.30_@opentelemetry+api@1.9.0_schh4asta4bbq2ii25cs3g5pay/node_modules/next-auth/next/index.js:18:19)\n' +
' at async p (/app/web/.next/server/pages/api/auth/[...nextauth].js:1:5724)\n' +
' at async /app/node_modules/.pnpm/@sentry+nextjs@8.55.0_@opentelemetry+context-async-hooks@1.30.1_@opentelemetry+api@1.9.0__@op_67qfrb3xwbsrn7n5hemqwejyci/node_modules/@sentry/nextjs/build/cjs/common/pages-router-instrumentation/wrapApiHandlerWithSentry.js:89:28\n' +
' at async K (/app/node_modules/.pnpm/next@14.2.30_@babel+core@7.24.3_@opentelemetry+api@1.9.0_@playwright+test@1.47.2_babel-plugin_pz5jczsrwmn5olwahiiemkgmoy/node_modules/next/dist/compiled/next-server/pages-api.runtime.prod.js:20:16881)\n' +
' at async U.render (/app/node_modules/.pnpm/next@14.2.30_@babel+core@7.24.3_@opentelemetry+api@1.9.0_@playwright+test@1.47.2_babel-plugin_pz5jczsrwmn5olwahiiemkgmoy/node_modules/next/dist/compiled/next-server/pages-api.runtime.prod.js:20:17520)\n' +
' at async NextNodeServer.runApi (/app/node_modules/.pnpm/next@14.2.30_@babel+core@7.24.3_@opentelemetry+api@1.9.0_@playwright+test@1.47.2_babel-plugin_pz5jczsrwmn5olwahiiemkgmoy/node_modules/next/dist/server/next-server.js:603:9)',
name: 'Error'
}
